src/Microsoft.AspNetCore.Server.Kestrel/KestrelServerOptions.cs

```diff
@@ -6,26 +6,60 @@
 
 namespace Microsoft.AspNetCore.Server.Kestrel
 {
+    /// <summary>
+    /// Provides programmatic configuration of Kestrel-specific features.
+    /// </summary>
     public class KestrelServerOptions
     {
+        /// <summary>
+        /// Enables the UseKestrel options callback to resolve and use services registered by the application during startup.
+        /// Typically initialized by <see cref="Hosting.WebHostBuilderKestrelExtensions.UseKestrel(Hosting.IWebHostBuilder, Action{KestrelServerOptions})"/>.
+        /// </summary>
         public IServiceProvider ApplicationServices { get; set; }
 
+        /// <summary>
+        /// Gets or sets an <see cref="IConnectionFilter"/> that allows each connection <see cref="System.IO.Stream"/>
+        /// to be intercepted and transformed.
+        /// Configured by the <c>UseHttps()</c> and <see cref="Hosting.KestrelServerOptionsConnectionLoggingExtensions.UseConnectionLogging(KestrelServerOptions)"/>
+        /// extension methods.
+        /// </summary>
+        /// <remarks>
+        /// Defaults to null.
+        /// </remarks>
         public IConnectionFilter ConnectionFilter { get; set; }
 
+        /// <summary>
+        /// Equivalent to <see cref="System.Net.Sockets.Socket.NoDelay"/>. Set to false to enable Nagle's algorithm.
+        /// </summary>
+        /// <remarks>
+        /// Defaults to true.
+        /// </remarks>
         public bool NoDelay { get; set; } = true;
 
         /// <summary>
         /// Gets or sets whether the <c>Server</c> header should be included in each response.
         /// </summary>
+        /// <remarks>
+        /// Defaults to true.
+        /// </remarks>
         public bool AddServerHeader { get; set; } = true;
 
         /// <summary>
         /// The amount of time after the server begins shutting down before connections will be forcefully closed.
-        /// By default, Kestrel will wait 5 seconds for any ongoing requests to complete before terminating
+        /// Kestrel will wait for the duration of the timeout for any ongoing requests to complete before terminating
```

Comment:
I think that it might be worth noting that no new connections will be accepted during the timeout.